### PR TITLE
Upgrade to latest prodcon build and add targeting packs for net46 support

### PR DESCRIPTION
**Pull request by crummel**
Testing build with net46 changes

### CODE REVIEW

**Comment by crummel** [comment]
@dotnet-bot test this please

**Comment by crummel** [comment]
CI seems messed up, closing and reopening.

**Comment by crummel** [comment]
@dseefeld @dagood This should fix the net46 issues for us for now.

File: patches/msbuild/0001-Add-TargetingPack-and-enable-net46-builds-for-source.patch [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+   <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">
+     <PackageReference Include="Microsoft.VisualStudio.Setup.Configuration.Interop" Version="$(VisualStudioSetupInteropVersion)" />
+@@ -982,4 +979,9 @@
+     -->
+     <Content Update="@(Content)" Pack="false" />
+   </ItemGroup>
++
++  <Target Name="CopyMscorlib" AfterTargets="Build">
++    <Copy SourceFiles="$(NuGetPackageRoot)\netstandard.library\2.0.0\build\netstandard2.0\ref\netstandard.dll" DestinationFolder="$(ArtifactsBinDir)/MSBuild/$(TargetFramework)/ref" />
++    <Copy SourceFiles="$(NuGetPackageRoot)\netstandard.library\2.0.0\build\netstandard2.0\ref\mscorlib.dll" DestinationFolder="$(ArtifactsBinDir)/MSBuild/$(TargetFramework)/ref" />
```
Comment (dagood) [inline patches/msbuild/0001-Add-TargetingPack-and-enable-net46-builds-for-source.patch]: What is this change about? I don't see it on https://github.com/NuGet/NuGet.Client/pull/2287.

Comment (crummel) [reply patches/msbuild/0001-Add-TargetingPack-and-enable-net46-builds-for-source.patch]: It's actually on the MSBuild PR: https://github.com/Microsoft/msbuild/pull/3389/files#diff-4d3fc7780500131817fceb95ddcea3feR983.  Copying mscorlib after the build means it doesn't get pulled into the references, it's a workaround for now.

**Comment by crummel** [comment]
Ah, I rebased your commit and got my patch commits in there as well.  I changed them back in my commits  so I'll squash when this gets merged so it doesn't show up as changing them and then reverting.  Thanks for the catch.

**Comment by dagood** [comment]
@dotnet-bot test OSX10.12 Release (java.nio.channels.ClosedChannelException)

**Comment by dagood** [comment]
@dotnet-bot test RHEL7.2 Tarball Release (I didn't see anything queued on https://ci.dot.net/job/dotnet_source-build/job/release_2.1/job/RHEL7.2_Tarball_Release/ ...)

**Comment by dagood** [comment]
Realized that the update misses the AspNetCore version and ProdCon feed updates. Filed https://github.com/dotnet/source-build/issues/608 to fix.

**Comment by dagood** [comment]
I'm working on batching up the rest of the 2.1.1 changes into a branch based on this PR so I can open the PR for those after this one gets in.

**Comment by dagood** [comment]
@dotnet-bot test OSX10.12 Release (java.nio.channels.ClosedChannelException)

**Comment by dagood** [comment]
@dotnet-bot test OSX10.12 Release (last try didn't seem to do anything.)

**Pull request merged**